pkg/v1/tkg/client/cluster.go

```diff
@@ -405,9 +418,20 @@ func (c *TkgClient) waitForCRS(options waitForAddonsOptions) error {
 	if err != nil {
 		return errors.Wrap(err, "error waiting for ClusterResourceSet object to be applied for the cluster")
 	}
+
 	return nil
 }
 
+// WaitForAddonsCorePackagesInstallation gets ClusterBootstrap and collects list of addons core packages, and monitors the kapp controller package installation in management cluster and rest of core packages installation in workload cluster
+func (c *TkgClient) WaitForAddonsCorePackagesInstallation(options waitForAddonsOptions) error {
+	clusterBootstrap, err := GetClusterBootstrap(options.regionalClusterClient, options.clusterName, options.namespace)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error while getting ClusterBootstrap object for workload cluster: %v", options.clusterName))
+	}
+	packages := GetCorePackagesFromCB(clusterBootstrap)
```

Comment:
nit: IMO `GetCorePackagesFromClusterBootstrap` might be more readable than `GetCorePackagesfromCB`

Reply:
agreed. done.